### DISCUSSION TITLE
Should LDK be so opinionated about `our_to_self_delay`?

### DISCUSSION THREAD

**Discussion by luckysori** [lightningdevkit/rust-lightning#2260] [General]
We have a situation where our node is happy to take on the risk of letting their channel counterparty claim their force-close funds after 12 blocks, but we are hitting this error on channel setup:

https://github.com/lightningdevkit/rust-lightning/blob/56b0c9683864d6860aa117b80470eda64e104854/lightning/src/ln/channel.rs#L1223-L1225

where `BREAKDOWN_TIMEOUT` is set to 144 blocks.

Whilst I understand that it's safe to enforce this at the library level, it's also limiting. Does it never make sense to go below 144 blocks? The number itself is arbitrary, which leads me to think that the library shouldn't have an opinion on this.

**Comment by TheBlueMatt** [comment]
I'm curious what your situation is - the 144 constant is designed to include the required time for a user to notice something is wrong, report an issue, and work with developers to resolve it before the timer runs out. For that a day is even relatively short.

**Comment by ariard** [comment]
@luckysori 

> Does it never make sense to go below 144 blocks?

There is clearly a case to go under 144 blocks to reduce the timevalue cost of the server liquidity in face of unreliable counterparty with a high-rate of non-cooperative force-close. The downside is in case of security issues, there is very low-odd we can release a security patch to fix the issues in let’s say 12 blocks. Where the we is defined at large with the Lightning folks/Bitcoin Core folks for channel compatibility and transaction-relay propagation compatibility issues. The hard things about hard things, I think the understanding of the whole matrix of liquidity/security risks is still in early days across the ecosystem, so for now I would rather recommend to modify BREAKDOWN_TIMEOUT for the channel acceptance check, especially if you’re already maintaining your own fork.
